### DISCUSSION TITLE
Is there a way to make loading.js work with client components?

### DISCUSSION THREAD

**Discussion by adleviton** [vercel/next.js#57379] [App Router] [answered]
It seems that loading.js only works when the component being exported is an async function. And async functions cannot be used in any file with the 'use client' declaration. Does anyone have comments on this?

```

//app/page.js
import Test from ../lib/test.js

export default async function Home() {
  return (
    <Test></Test>
  )
}

//test.js, server component
//loading.js appears no problem
export default async function Test() {

  await new Promise(resolve => setTimeout(resolve, 5000))
  
  return (
  <div>My Component</div>
  )
}

//test.js, client component
//loading.js does not appear
export default function Test() {

  /*something that takes a few seconds to hydrate/render*/
  
  return (
  <div>My Component</div>
  )
}

```

**Comment by mass2527** [comment, marked as answer]
If you create a `loading.js` file, Next.js will automatically wrap the `Home` component using the `Suspense` component and pass the `<Loading />` component from `loading.js` to the `fallback` prop, like this:

```tsx
<Suspense fallback={<Loading />}>
   <Home>
      <Test />
   </Home>
</Suspense>
```

To make loading.js work as described, Suspense should be in effect. Just so you know, only data sources enabled with Suspense will activate the Suspense component. The rendering code itself does not activate the Suspense component.

For more information, please refer to the Note on [Suspense usage](https://react.dev/reference/react/Suspense#usage).

**Reply by adleviton** [reply]
Thank you for this. My actual situation is a bit more complex and I needed to handle the loading screen with `useEffect`, but otherwise, the info you shared explains what's happening and points me in the right direction for more typical situations.
